web/lib/components/common/SupabaseAlert.jsx

```diff
@@ -0,0 +1,26 @@
+import { Alert } from './Alert'
+import { useTranslation } from '../../i18n'
+
+export const SupabaseAlert = function ({ error, defaultMessage }) {
+  const { _e } = useTranslation()
+
+  switch (error.message) {
+    case 'MAX_ALLOWED_SETS_CREATED': {
+      return (
+        <Alert variant="danger">
+          {_e('flashcardset.error.exceedMaxNumberOfSets')}
+        </Alert>
+      )
+    }
+    case 'UNAUTHORIZED_MODIFY_FLASHCARDS': {
+      return (
+        <Alert variant="danger">
+          {_e('flashcard.errors.onlyCreatorIsAllowedToAddModifyFlashcards')}
+        </Alert>
+      )
+    }
+    default: {
+      return <Alert variant="danger">{defaultMessage}</Alert>
+    }
+  }
```

Comment:
A map is less redundant?

```jsx
const errMap = new Map([
  ['MAX_ALLOWED_SETS_CREATED', 'flashcardset.error.exceedMaxNumberOfSets'],
  ...
])

return <Alert variant="danger">{
  errMap.has(error.message) ? _e(errMap.get(error.message)) : defaultMessage
}</Alert>
```

Reply:
Or better by having this `flashcardset.error.MAX_ALLOWED_SETS_CREATED` as an i18n message.